PIL/GifImagePlugin.py

```diff
@@ -439,13 +439,9 @@ def _get_local_header(fp, im, offset, flags):
 
             # adjust the transparency index after optimize
             if len(used_palette_colors) < 256:
-                for i in range(len(used_palette_colors)):
-                    if used_palette_colors[i] == transparency:
-                        transparency = i
-                        transparent_color_exists = True
-                        break
-                    else:
-                        transparent_color_exists = False
+                transparent_color_exists = transparency in used_palette_colors
+                if transparent_color_exists:
+                    transparency = used_palette_colors.index(transparency)
```

Comment:
This refactored line isn't covered by tests:
https://travis-ci.org/python-pillow/Pillow/jobs/182957414#L4912

Is it possible to test it?